src/runtime/String.h

```diff
@@ -77,6 +77,14 @@ class RopeString;
 class StringView;
 class Context;
 
+class StringWriteOption {
+public:
+    enum {
```

Comment:
IMHO `enum class` is suitable for this case instead of declaring a class.
Other looks good to me.

Reply:
Generally `enum class` would be better but `options` is rather kind of bit flags and probably could be combined. Here is definition write options from v8.h:
`    enum WriteOptions {
      NO_OPTIONS = 0,
      HINT_MANY_WRITES_EXPECTED = 1,
      NO_NULL_TERMINATION = 2,
      PRESERVE_ONE_BYTE_NULL = 4,
      REPLACE_INVALID_UTF8 = 8
    };`
For `enum class` operators `|` and `&` should be defined. Value returned by `operator|` will be out of defined enum values.
What do you think about leaving it as is?

Reply:
For bit operations on enum class values, we may have to explicitly cast each type ahead of operation and it would be inconvenient. So, IMO your appoach is fine to me.